pandas/core/indexes/base.py

```diff
@@ -2502,10 +2522,11 @@ def symmetric_difference(self, other, result_name=None):
         right_diff = other.values.take(right_indexer)
 
         the_diff = _concat._concat_compat([left_diff, right_diff])
-        try:
-            the_diff = sorting.safe_sort(the_diff)
-        except TypeError:
-            pass
+        if sort:
+            try:
+                the_diff = sorting.safe_sort(the_diff)
```

Comment:
so we generally want to use safe_sort if we can (and it seems we are using a safe_sort, sort_values, and np.sort) in various places. let's start (as i mentioned above) with a single function that can sort (you may have to have some ugly if/else logic inside it to handle the differnt types of objects, e.g. ndarrays & Index)

Reply:
It *may* pay to do a pre-cursor PR for this factoring out.